[강예지] Refactor : 코드리뷰 반영(코드 스타일 수정)

Kangyejiii opened the pull request:
wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
Refactor : 코드리뷰 반영

## 수정 사항들 자세한 내용
- map()을 이용해서 카드 리스트  리턴하게 했습니다.
- 추가 기능인 검색도 구현했습니다.
- [Fix] 대문자로 검색했을 때 결과가 나오지 않아서 코드 수정했습니다.(8/12)
- 가독성을 위해 코드 스타일 수정( 띄어쓰기 )(8/13)
- 줄일 수 있는 코드 수정 (`result.length===0` -> `!result.length`)

## 기타 질문 및 특이 사항

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -14,33 +14,54 @@ import "./index.css";
     // 필터링 로직
+    const { monsters, userInput } = this.state
+    let filtered = monsters.filter(
+      (monster) => monster.name.toLowerCase().includes(userInput),
```
더욱 더 정확한 비교를 위해 userInput도 소문자 형태로 변환해주세요~

Kangyejiii committed:
Add : 카드 (이름,이메일) 정보 출력, 검색기능 구현
- map()을 이용해서 카드 리스트  리턴하게 했습니다.
- 추가 기능인 검색도 구현했습니다.

Kangyejiii committed:
Fix : 검색 기능  코드 수정
- 입력칸에서 대문자를 포함해서 검색할 시 결과가 안나와서 코드 수정

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Components/Card/Card.js:
```diff
@@ -22,11 +22,11 @@ class Card extends Component {
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.name}</h2>
+        <p>{this.props.email}</p>
```
👍 👍 👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Components/CardList/CardList.js:
```diff
@@ -11,7 +11,10 @@ import "./CardList.scss";
   render() {
-    return <div className="card-list"></div>;
+    return <div className="card-list">
+      
```
불필요한 빈줄입니다! 없애주세요 😊

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Components/CardList/CardList.js:
```diff
@@ -11,7 +11,10 @@ import "./CardList.scss";
-    return <div className="card-list"></div>;
+    return <div className="card-list">
+      
+      {this.props.monsters.map((el)=>{return <Card key={el.id} id={el.id} name={el.name} email={el.email}></Card>})}
```
- ES6 의 arrow function 은 함수의 body 에서 단순히 return 만 한다면 생략할 수 있는 부분들이 있습니다!
- JSX element 는 자식 tag 혹은 내용이 없다면, self-closing tag 를 활용하여 태그를 닫아줄 수 있습니다 😎

적용해서 push 해주세요! 😄

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Monsters.js:
```diff
@@ -30,15 +30,38 @@ class Monsters extends Component {
+  componentDidMount() {
+    fetch("https://jsonplaceholder.typicode.com/users")
+    .then(res=>res.json())
+    .then(res=>this.setState({monsters : res}));
```
fetch 함수 배우신대로 정말 잘 적용하셨네요! 😉

하지만, 깔끔하고 잘 작동하는 로직 외에도 정말 중요한 점이 있다면, 코드 스타일입니다.
가독성이 좋은 코드는 협업하는 팀원들로 하여금 내 코드를 읽고 이해하는데 엄청나게 많은 시간을 단축시켜주기 때문이죠 😎

그래서 해당 코드를 아래처럼 스타일링 해주실 수 있습니다!
```suggestion
    fetch("https://jsonplaceholder.typicode.com/users")
      .then(res => res.json())
      .then(res => this.setState({ monsters: res }));
```

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Monsters.js:
```diff
@@ -30,15 +30,38 @@ class Monsters extends Component {
+
+  handleChange = (e) => {
+    this.setState({userInput: e.target.value});
+    }
```
여기도 아래처럼 closing bracket 위치 변경해주세요!
```suggestion
  handleChange = (e) => {
    this.setState({userInput: e.target.value});
  }
```

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Monsters.js:
```diff
@@ -30,15 +30,38 @@ class Monsters extends Component {
+    }
+
+    handleSearch = () => {
+      let result=this.state.monsters.filter((el)=>(el.name.toLowerCase().includes(this.state.userInput.toLowerCase())));
```
filter 함수 잘 사용하셨네요! Goood! 😄

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Monsters.js:
```diff
@@ -30,15 +30,38 @@ class Monsters extends Component {
+    handleSearch = () => {
+      let result=this.state.monsters.filter((el)=>(el.name.toLowerCase().includes(this.state.userInput.toLowerCase())));
+      
+      if(!this.state.userInput.trim("")||result.length===0) {
```
- 오오.. 정확한 비교를 위해 `trim()` 함수를 이용해 불필요한 space 를 제거해주는 부분 너무 인상깊네요! 👍 👍 👍 
- `result.length` 의 값이 `0` 인지 확인도 잘해주셨는데, `0` 은 falsy value 라서 `!result.length` 로 줄여줄 수도 있겠네요!

마찬가지로 코드스타일 조금만 지적하겠습니다 ㅎㅎㅎ
```suggestion
      if (!this.state.userInput.trim() || result.length === 0) {
```

jamessoun93 commented on src/Assignments/Monsters/students/11th/Kangyejiii/Monsters.js:
```diff
@@ -30,15 +30,38 @@ class Monsters extends Component {
-        {/* <CardList monsters=몬스터리스트 /> */}
-      </div>
+        <SearchBox handleChange={this.handleChange}/>
+        {result?  <CardList  monsters={result}/> : <div><h2>검색해보세요:-D</h2></div> }
```
삼항연산자 와우! 🥳 🥳 🥳 

이 부분도 코드스타일 조금만... 지적하겠습니다 😊
```suggestion
        {result ? <CardList monsters={result} /> : (
          <div>
            <h2>검색해보세요:-D</h2>
          </div>
        }
```

jamessoun93 reviewed (changes requested):
예지님! 리뷰 남겨드렸습니다 😎

반영해서 push 해주세요!

Kangyejiii committed:
Refactor : 코드리뷰 반영
- 가독성을 위해 코드 스타일 수정( 띄어쓰기 )
- 줄일 수 있는 코드 수정 (`result.length===0` -> `!result.length`)

jamessoun93 commented:
LGTM 👍👍👍
수고하셨습니다!!
